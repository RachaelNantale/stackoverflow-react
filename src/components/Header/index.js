import React from 'react';
import './navbar.scss';


const Header = () => (
  <div className="header">
    <nav className="navbar nav navbar-expand-lg  navbar-dark bg-dark">
      <a className="navbar-brand logo" href="/">
       StackOverflow-lite
      </a>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarTogglerDemo02"
        aria-controls="navbarTogglerDemo02"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>

      <div className="collapse navbar-collapse" id="navbarTogglerDemo02">
        <ul className="navbar-nav mr-auto mt-1 mt-lg-0 nav-1">
          <a href="/post_question" className="nav-link active">Create Question </a>
        </ul>
        <ul className="navbar-nav mr-2 mt-2 mt-lg-0 nav-2">
          <a href="/login" className="nav-link active">Login</a>
          <a href="/signup" className="nav-link active">Signup </a>
        </ul>

        <form className="form-inline my-2 my-lg-0 ">
          <input className="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search" />
          <div className="input-group-btn">
            <button className="btn btn-outline-primary my-2 my-sm-0" type="submit">Search</button>
          </div>
        </form>
      </div>
    </nav>
  </div>
);

export default Header;
