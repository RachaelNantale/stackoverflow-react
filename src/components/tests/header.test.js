import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

it('renders the signup component', () => {
  const HomeComponent = shallow(<Header />);
  expect(HomeComponent).toMatchSnapshot();
});
