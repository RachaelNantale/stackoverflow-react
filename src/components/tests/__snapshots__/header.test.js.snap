// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the signup component 1`] = `
<div
  className="header"
>
  <nav
    className="navbar nav navbar-expand-lg  navbar-dark bg-dark"
  >
    <a
      className="navbar-brand logo"
      href="/"
    >
      StackOverflow-lite
    </a>
    <button
      aria-controls="navbarTogglerDemo02"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarTogglerDemo02"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarTogglerDemo02"
    >
      <ul
        className="navbar-nav mr-auto mt-1 mt-lg-0 nav-1"
      >
        <a
          className="nav-link active"
          href="/post_question"
        >
          Create Question 
        </a>
      </ul>
      <ul
        className="navbar-nav mr-2 mt-2 mt-lg-0 nav-2"
      >
        <a
          className="nav-link active"
          href="/login"
        >
          Login
        </a>
        <a
          className="nav-link active"
          href="/signup"
        >
          Signup 
        </a>
      </ul>
      <form
        className="form-inline my-2 my-lg-0 "
      >
        <input
          aria-label="Search"
          className="form-control mr-sm-2"
          placeholder="Search"
          type="search"
        />
        <div
          className="input-group-btn"
        >
          <button
            className="btn btn-outline-primary my-2 my-sm-0"
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </div>
  </nav>
</div>
`;
